### ISSUE TITLE
TooManyArguments does not support methods in issueHandlers

### ISSUE THREAD

**Issue by GTony** [vimeo/psalm#10070]
Since 5.14 following the changes in #9981 it's not longer possible to pass a method to `referencedFunction`

As of version 5.13.1 our `psalm.xml` looked as follow.

```
<issueHandlers>
    <TooManyArguments>
        <errorLevel type="suppress">
            <referencedFunction name="Doctrine\DBAL\Query\QueryBuilder::select"/>
            ...
        </errorLevel>
    </TooManyArguments>
</issueHandlers>
```

But now running psalm will return this error as per the recent changes.
```
Invalid referencedFunction doctrine\dbal\query\querybuilder::select
```

As it now stands it's impossible to suppress TooManyArguments for anything other than basic function.

Should TooManyArguments support `referencedMethod` or should the regex on referencedFunction also support methods?

It's also possible I'm entirely in the wrong and I shouldn't suppress TooManyArguments error for methods.

To be noted, I am aware of the existence of a plugin to help psalm with some of Doctrine's issues, but this is not the problem right here.

**Comment by weirdan** [comment]
> Should TooManyArguments support `referencedMethod` 

It should.

**Comment by weirdan** [comment]
> It should.

From a user standpoint, that is. Not exactly easy to implement, given how we filter issues.

Anyway, this seems like an unintentional BC break, and I'll be looking to fix it.

/cc: @orklah @kkmuffme

**Comment by weirdan** [comment]
@GTony meanwhile, you can work this around by converting the filter into a PCRE regular expression:

```xml
            <referencedFunction name="@Doctrine\\DBAL\\Query\\QueryBuilder::select@"/>
```

**Comment by weirdan** [comment]
The fix is available in [5.14.1](https://github.com/vimeo/psalm/releases/tag/5.14.1).
